pulsar-client/src/test/java/org/apache/pulsar/client/impl/MessageImplTest.java

```diff
@@ -71,4 +78,48 @@ public void testGetProducerNameAssigned() {
         assertEquals("test-producer", msg.getProducerName());
     }
 
+    @Test
+    public void testGetProducerDataAssigned() {
+        MessageMetadata.Builder builder = MessageMetadata.newBuilder()
+                .setProducerName("test-producer");
+
+//        ByteBuffer payload = ByteBuffer.wrap(new byte[0]);
+        AvroSchema<SchemaTestUtils.Foo> fooSchema = AvroSchema.of(SchemaDefinition.<SchemaTestUtils.Foo>builder().withPojo(SchemaTestUtils.Foo.class).build());
+        AvroSchema<SchemaTestUtils.Bar> barSchema = AvroSchema.of(SchemaDefinition.<SchemaTestUtils.Bar>builder().withPojo(SchemaTestUtils.Bar.class).build());
+
+        Schema<KeyValue<SchemaTestUtils.Foo, SchemaTestUtils.Bar>> keyValueSchema = Schema.KeyValue(fooSchema, barSchema);
+        SchemaTestUtils.Foo foo = new SchemaTestUtils.Foo();
+        foo.setField1("field1");
+        foo.setField2("field2");
+        foo.setField3(3);
+        SchemaTestUtils.Bar bar = new SchemaTestUtils.Bar();
+        bar.setField1(true);
+        Map<String, String> properties = Maps.newHashMap();
+
+        // // Check kv.encoding.type default, not set value
+        byte[] encodeBytes = keyValueSchema.encode(new KeyValue(foo, bar));
+        MessageImpl<KeyValue<SchemaTestUtils.Foo, SchemaTestUtils.Bar>> msg = MessageImpl.create(builder, ByteBuffer.wrap(encodeBytes), keyValueSchema);
+        KeyValue<SchemaTestUtils.Foo, SchemaTestUtils.Bar> keyValue = msg.getValue();
+        assertEquals(keyValue.getKey(), foo);
+        assertEquals(keyValue.getValue(), bar);
+
+        // Check kv.encoding.type INLINE
+        properties.put("kv.encoding.type", "INLINE");
+        keyValueSchema.getSchemaInfo().setProperties(properties);
+        encodeBytes = keyValueSchema.encode(new KeyValue(foo, bar));
+        msg = MessageImpl.create(builder, ByteBuffer.wrap(encodeBytes), keyValueSchema);
+        keyValue = msg.getValue();
+        assertEquals(keyValue.getKey(), foo);
+        assertEquals(keyValue.getValue(), bar);
+
+        // Check kv.encoding.type SPRAERATE
+        properties.put("kv.encoding.type", "SEPARATED");
+        keyValueSchema.getSchemaInfo().setProperties(properties);
```

Comment:
How does keyValueSchema know the encoding type is `SEPARATED`?  What you did here is just adding properties. How is the `keyValueEncodingType` field changed in keyValueSchema?